src/EpicWithSpecialHeader/index.tsx

```diff
@@ -52,34 +48,62 @@ const headerStyles = {
     `,
 };
 
+type HeaderSectionProps = {
+    authoredEpicImageUrl?: string;
```

Comment:
should all these fields be mandatory in this component?

Reply:
I think probably for this component some of these should be required (i.e. if it'd look odd to render the header with no byline name, then we should make it required). I think TypeScript will force us to do this, but in the `Epic` component, we should only render `HeaderSection` if the required props are available.

Reply:
Is it possible to nest the mandatory fields under an optional field in the main epic props type? E.g.
```
type BrazeMessageProps = {
  buttonText?: string;
  ...
  authorHeader?: {
    authoredEpicImageUrl: string;
    authoredEpicImageAltText: string;
    authoredEpicHeader: string;
    authoredEpicBylineName: string;
    authoredEpicBylineCopy1: string;
    authoredEpicBylineCopy2: string;
  }
```

Then `HeaderSection` doesn't have to deal with optional fields.

Reply:
Ah yep that's nice 👍 (but remember that these will all be coming through as a flat structure of key/value pairs from Braze, so we'll have to do the nesting here)

Reply:
I think I'd prefer the props to be flat? Hierarchical data (in my view) adds complexity where it's not essential and may trip up those who have to maintain components like this going forward?

Reply:
Sure. My original point was that if fields are necessary to render a component, it's simpler + more correct for that component to receive them as non-optional. But since braze doesn't support nesting there's probably not much to be gained by nesting here

Reply:
Updated to check the imageUrl, altText and bylineName fields have been completed.

Reply:
> Updated to check the imageUrl, altText and bylineName fields have been completed.

Great 👍  I think the types here should probably reflect that, what do you think? I.e. right now the `HeaderSectionProps` type says that `authoredEpicImageUrl` is optional, but in the JSX we use it as though it's non-optional (using it as the source for a `img` tag for example.

Reply:
Done

Reply:
I'm confused about why some of these props are optional and some not - it looks like they're all being rendered?

Reply:
Pushed an update - does that address your concern?